include/boost/uuid/detail/random_provider.hpp

```diff
@@ -0,0 +1,63 @@
+//
+// Copyright (c) 2017 James E. King III
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at
+//   http://www.boost.org/LICENCE_1_0.txt)
+//
+// Platform-specific random entropy provider
+//
+
+#ifndef BOOST_UUID_DETAIL_RANDOM_PROVIDER_HPP
+#define BOOST_UUID_DETAIL_RANDOM_PROVIDER_HPP
+
+#include <boost/core/noncopyable.hpp>
+#include <boost/cstdint.hpp>
+#include <boost/limits.hpp>
+#include <boost/static_assert.hpp>
+#include <boost/uuid/entropy_error.hpp>
+
+#include <boost/uuid/detail/random_provider_detect_platform.hpp>
+#include BOOST_UUID_RANDOM_PROVIDER_BASE_INCLUDE()
```

Comment:
It's better to just include the appropriate platform header from `random_provider_detect_platform.hpp`, renaming it into `random_provider_base.hpp` in the process, because the above confuses dependency scanners; they can't see through the macro.

Reply:
I needed platform detection separate from loading the implementation to make the mocking work properly.  I can expand that out so the macro isn't there and put additional preprocessor branches based on the `BOOST_UUID_RANDOM_PROVIDER_*` definitions made by the detection logic in that header.

Reply:
Yes, it's better to spell it out with #ifdefs instead of a macro include.